src/ORM/Search/ElementalSiteTreeSearchContext.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace DNADesign\Elemental\ORM\Search;
+
+use DNADesign\Elemental\Extensions\ElementalPageExtension;
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\CMS\Search\SiteTreeSearchContext;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\ORM\DataQuery;
+use SilverStripe\ORM\Filters\ExactMatchFilter;
+
+class ElementalSiteTreeSearchContext extends SiteTreeSearchContext
+{
+    use Configurable;
+
+    /**
+     * If false, the custom search against elemental block content will not be performed.
+     * Set this to false for best performance.
+     */
+    private static bool $search_for_term_in_content = true;
+
+    /**
+     * Whether to render elements with templates when doing a CMS SiteTree search.
+     * Setting this to false will give a big boost to performance with minimal impact to the user experience.
+     */
+    private static bool $render_elements = true;
+
+    protected function generalSearchAcrossFields(string|array $searchPhrase, DataQuery $subGroup, array $searchableFields): void
+    {
+        parent::generalSearchAcrossFields($searchPhrase, $subGroup, $searchableFields);
+
+        if (static::config()->get('search_for_term_in_content') === false) {
+            return;
+        }
+
+        if (!is_array($searchPhrase)) {
+            $searchPhrase = [$searchPhrase];
+        }
+
+        $pageIDs = [];
+        // The same extension can't be applied to the multiple classes in the same hierarchy
+        // without causing a host of problems, so we can be confident that we're not getting any double ups here.
+        $pageClassesWithExtension = ClassInfo::classesWithExtension(ElementalPageExtension::class, SiteTree::class, true);
+        // Get a list of classes that can't have elemental blocks despite having the extension to reduce the amount of
+        // records we're filtering through needlessly
+        $ignoredClasses = Config::forClass(ElementalPageExtension::class)->get('ignored_classes');
+        foreach ($ignoredClasses as $class) {
+            $ignoredClasses = array_merge($ignoredClasses, ClassInfo::subclassesFor($class, false));
+        }
```

Comment:
This part and the `filter` and `exclude` calls are new, so we don't filter through records that just aren't relevant at all.
Everything else is the same which some minor refactoring to work in this method as opposed to the old one.